### PR TITLE
Bump a couple deps; proactively deal with Throwables deprecation

### PR DESCRIPTION
**Pull request by mrhota**

### CODE REVIEW

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadSingleWriteLock.java [hunk 1]
```diff
@@ -55,11 +54,8 @@ public boolean acquireWriteLock(String ownerId) {
                 Thread.sleep(250);
             }
         } catch (InterruptedException e) {
-            Throwables.propagate(e);
+            throw new RuntimeException(e);
```
Comment (cschellenger) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadSingleWriteLock.java]: I realize it's a change in behavior, but you should probably call `Thread.currentThread().interrupt()`, as above when catching `InterruptedException`

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadSingleWriteLock.java [hunk 2]
```diff
@@ -83,11 +79,8 @@ public boolean acquireReadLock(String ownerId) {
...
```
Comment (cschellenger) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadSingleWriteLock.java]: Same. Re-interrupt.

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/cache/RedisHashCacheTest.java [hunk 3]
```diff
@@ -305,7 +304,7 @@ public TestContainer apply(@Nullable String s) {
                 try {
                     return queue.take();
                 } catch (InterruptedException e) {
-                    throw Throwables.propagate(e);
+                    throw new RuntimeException(e);
```
Comment (cschellenger) [inline rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/cache/RedisHashCacheTest.java]: Interrupt

File: rdbi-parent/pom.xml [hunk 4]
```diff
@@ -91,12 +91,12 @@
...
                 <groupId>com.google.code.findbugs</groupId>
                 <artifactId>jsr305</artifactId>
-                <version>2.0.1</version>
+                <version>3.0.1</version>
             </dependency>
             <dependency>
                 <groupId>joda-time</groupId>
                 <artifactId>joda-time</artifactId>
-                <version>2.9.4</version>
+                <version>2.9.9</version>
```
Comment (gjesse) [inline rdbi-parent/pom.xml]: what would it take to just drop this dependency? i've been slowly removing it in favor of java8 date utils.

Comment (mrhota) [reply rdbi-parent/pom.xml]: lololol now you're talkin'!

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadSingleWriteLock.java (hunk 1 above)
Comment (mrhota) [reply rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadSingleWriteLock.java]: hmm let's see what happens!

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/cache/RedisHashCacheTest.java (hunk 3 above)
Comment (mrhota) [reply rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/cache/RedisHashCacheTest.java]: this is in a unit test; i'm inclined to leave it as an exception.

**Comment by mrhota** [comment]
@gjesse removed joda!

**Pull request merged**